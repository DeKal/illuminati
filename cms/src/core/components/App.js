import React, { Fragment } from 'react'
import clsx from 'clsx'
import PropTypes from 'prop-types'
import NavBar from 'core/containers/NavBar'
import Drawer from 'core/containers/Drawer'
import PageSwitch from 'core/components/PageSwitch'

const App = ({ classes, open }) => (
  <Fragment>
    <Drawer />
    <main
      className={clsx(classes.content, {
        [classes.contentShift]: open
      })}
    >
      <NavBar />
      <PageSwitch />
    </main>
  </Fragment>
)

export default App

App.propTypes = {
  classes: PropTypes.object,
  open: PropTypes.bool
}
