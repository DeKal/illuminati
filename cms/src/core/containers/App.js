import React from 'react'
import { makeStyles } from '@material-ui/core/styles'
import { connect } from 'react-redux'
import App from 'core/components/App'

const useStyles = makeStyles(theme => ({
  content: {
    transition: theme.transitions.create('margin', {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen
    }),
    marginLeft: 0,
    [theme.breakpoints.down('sm')]: {
      width: '100%'
    }
  },
  contentShift: {
    transition: theme.transitions.create('margin', {
      easing: theme.transitions.easing.easeOut,
      duration: theme.transitions.duration.enteringScreen
    }),
    marginLeft: theme.drawerWidth
  }
}))

const AppContainer = props => <App classes={useStyles()} {...props} />

const mapStateToProps = state => ({
  open: state.drawer.open
})

export default connect(mapStateToProps)(AppContainer)
