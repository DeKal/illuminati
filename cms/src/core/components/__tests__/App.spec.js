import * as React from 'react'
import { shallow } from 'enzyme'
import App from 'core/components/App'

const classes = {
  content: 'open',
  contentShift: 'shift'
}
describe('App', () => {
  it('should render App with open drawer', function() {
    const wrap = shallow(<App classes={classes} open />)
    expect(wrap).toMatchSnapshot()
  })

  it('should render App with close drawer', function() {
    const wrap = shallow(<App classes={classes} open={false} />)
    expect(wrap).toMatchSnapshot()
  })
})
