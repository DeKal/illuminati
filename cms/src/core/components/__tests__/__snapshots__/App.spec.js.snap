// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render App with close drawer 1`] = `
<Fragment>
  <Connect(DrawerContainer) />
  <main
    className="open"
  >
    <Connect(withRouter(NavBarContainer)) />
    <PageSwitch />
  </main>
</Fragment>
`;

exports[`App should render App with open drawer 1`] = `
<Fragment>
  <Connect(DrawerContainer) />
  <main
    className="open shift"
  >
    <Connect(withRouter(NavBarContainer)) />
    <PageSwitch />
  </main>
</Fragment>
`;
